File: lei_lds2122_grupo2-main/Frontend/src/app/viewModels/requests/CommentRouteVM.ts
import { toJSONString } from "../interfaces/toJSONString";

/**
 * View Model class for a comment request
 */
export class CommentRouteVM implements toJSONString {

    /**
     * Construct's an instance of {@link CommentRouteVM}
     * 
     * @param comment the comment
     */
    constructor(
        public comment: string
    ) { }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    public toJSONString(): string {
        return JSON.stringify({ "comment": this.comment })
    }
}

File: lei_lds2122_grupo2-main/Backend/Data/ViewModels/Request/CommentRouteVM.cs
namespace Backend.Data.ViewModels.Request
{
    /// <summary>
    /// View Model class for commnet route request
    /// </summary>
    public class CommentRouteVM
    {
        public string Comment{ get; set; }

        /// <summary>
        /// Class' contructor method
        /// </summary>
        /// <param name="comment">Commnet to add</param>
        public CommentRouteVM(string comment)
        {
            this.Comment = comment;
        }
    }
}